app/javascript/mastodon/components/federation_icon.tsx

```diff
@@ -0,0 +1,57 @@
+import { defineMessages, useIntl } from 'react-intl';
+
+import LockIcon from '@/material-icons/400-24px/lock.svg?react';
+import PublicIcon from '@/material-icons/400-24px/public.svg?react';
+import QuietTimeIcon from '@/material-icons/400-24px/quiet_time.svg?react';
+import type { StatusFederation } from 'mastodon/models/status';
+
+import { Icon } from './icon';
+
+const messages = defineMessages({
+  federation_full_short: {
+    id: 'federation.federation_full.short',
+    defaultMessage: 'Fediverse',
+  },
+  federation_local_short: {
+    id: 'federation.federation_local.short',
+    defaultMessage: 'Home server',
+  },
+  federation_limited_short: {
+    id: 'federation.federation_limited.short',
+    defaultMessage: 'Wider Community',
+  },
+});
+
+export const FederationIcon: React.FC<{ federation: StatusFederation }> = ({
```

Comment:
I opted to create a new component but perhaps we can pass in a new federation prop to the visibility_icon.tsx and use it there. I'd appreciate feedback in this regard.